Purchases/Networking/DNSChecker.swift

```diff
@@ -35,12 +37,18 @@ enum DNSChecker {
         return isBlockedURL(failedURL)
     }
 
-    static func blockedHostFromError(_ error: Error?) -> String? {
-        guard let failedURL = (error as NSError?)?.userInfo[NSURLErrorFailingURLErrorKey] as? URL else {
+    static func errorWithBlockedHostFromError(_ error: Error?) -> Error? {
+        guard let failedURL = (error as NSError?)?.userInfo[NSURLErrorFailingURLErrorKey] as? URL,
+                let nsError = error as NSError? else {
```

Comment:
we're doing the `error as NSError?` bit twice, perhaps inverting the order here would simplify things, i.e.: 
```suggestion
        guard let let nsError = error as NSError,
            let failedURL = nsError.userInfo[NSURLErrorFailingURLErrorKey] as? URL else {
```

also I removed the `?` because I think we'd want this to be non-null, right?

Reply:
Probably 😄 I'll check into it

Reply:
Yup. You're right. Fixed!